test/integration/mig_autoscaler/controls/mig_autoscaler.rb

```diff
@@ -0,0 +1,129 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+project_id = attribute('project_id')
+credentials_path = attribute('credentials_path')
+region     = attribute('region')
+
+ENV['CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE'] = File.absolute_path(
+  credentials_path,
+  File.join(__dir__, "../../../fixtures/mig/autoscaler"))
+
+expected_instances = 4
+expected_instance_groups = 1
+
+control "MIG" do
+  title "Autoscaling Configuration"
+
+  describe command("gcloud --project=#{project_id} compute instances list --format=json --filter='name~^mig-autoscaler*'") do
+    its(:exit_status) { should eq 0 }
+    its(:stderr) { should eq '' }
+
+    let!(:data) do
+      if subject.exit_status == 0
+        JSON.parse(subject.stdout)
+      else
+        {}
+      end
+    end
+
+    describe "number of instances" do
+      it "should be #{expected_instances}" do
+        expect(data.length).to eq(expected_instances)
+      end
+    end
+
+    describe "network tags" do
+      it "should include 'foo'" do
+        data.each do |instance|
```

Comment:
This still causes multiple assertions to be made within a single test (`it`). Additionally, the assertions are dynamically generated which means if `subject.exit_status != 0`, `data` will be an empty value and this test will contain no assertions and report no failure. Tests and assertions should be statically defined.

Reply:
@aaron-lane in this case, each instance in the MIG has a random suffix at the end (i.e. the instance name can't be statically determined). What's the best way to write this test? I might not be thinking about this in the right way...

Reply:
NVM, I can just directly get each instance via list indices 0, 1, 2, 3.

Reply:
@sosimon that's fair. Exposing the instance names as outputs and using the inspec-gcp resources could help to at least reduce the dynamic nature. Note that a control using the GCP backend will need to be run in a different Kitchen-Terraform verifier system than ones using the local backend with gcloud.

Example:

```ruby
describe google_compute_instance(project: attribute('project'),  zone: attribute('zone'), name: attribute('instance_1_name')) do
  # The documentation does not indicate the format of the tags attribute
  # Assuming it's equivalent to the JSON output
  its('tags') { should include 'items' => including('foo') }
end
```

Reply:
@aaron-lane I spent some time this morning to look at using the inspec-gcp resources, but:
1. the resources doesn't have all the properties that I would like to tests (e.g. the google_compute_region_instance_group_manager resource does not have autoscaler properties)
2. I have not figured out how to expose instance names as outputs from the MIG module (turns out it's not trivial)
Right now, I'm thinking about sticking with the local backend. But I will switch focus for now and revisit this later.

Reply:
I've split out the assertions and referencing each instance statically (by list index, not by name). While I think this satisfy the requirement that tests should be as static as possible, I'm not 100% sure this is the best way to write these tests.

Reply:
I have refactored the tests based on some additional feedback from Gary. I should have been testing the tags and labels in the instance_template test suite, not in this MIG test suite.